### DISCUSSION TITLE
Video's slow behind Reverse Proxy (Traefik)

### DISCUSSION THREAD

**Discussion by wardwygaerts** [immich-app/immich#8655] [Q&A]
I have Immich running behind Traefik. Everything is working fine, only video's are loaded very slow and do not play smoothly (or I need to wait untill the video is completely loaded). When I connect via my local IP, video's are playing instantly.
So I assume it's related to Traefik. I already played with the [buffering settings](https://doc.traefik.io/traefik/middlewares/http/buffering/), but no luck.

Or, are the video's to big to go over the internet? My upload speed is only 50 Mbps. Do I need to change the encoding settings in Immich? Can I test this for only one video?

**Comment by nicomerl** [comment]
My Immich is running behind Traefik with 10 Mbps Upload and videos are loading pretty fast. I guess its not a Problem with your upload speed. I did not change any encoding settings in Immich.

**Reply by wardwygaerts** [reply]
I also use Cloudflare for DNS and I switched from *proxied* to *DNS only* for the immich DNS entry, now video's are playing smoothy. So it's a Cloufslare thing... Can I change other setting in Cloudflare so I can still use the proxy feature?

**Comment by keesfluitman** [comment]
> I also use Cloudflare for DNS and I switched from _proxied_ to _DNS only_ for the immich DNS entry, now video's are playing smoothy. So it's a Cloufslare thing... Can I change other setting in Cloudflare so I can still use the proxy feature?

Ill check on it. Mine is also slow. using Swag (nginx) and cloudflare. Seems like it's completely downloading the videos before playing them.

**Reply by wardwygaerts** [reply]
I noticed the *DNS only* setting didn't solve it completely.
I now also changed the *target resolution* in de video transcoding settings to **720p** and changed the *transode policy* to **videos higher than target resolution or not in accepted format**.
After transcoding all video's, it seems to play all very smoothly.

**Comment by keesfluitman** [comment]
> I noticed the _DNS only_ setting didn't solve it completely. I now also changed the _target resolution_ in de video transcoding settings to **720p** and changed the _transode policy_ to **videos higher than target resolution or not in accepted format**. After transcoding all video's, it seems to play all very smoothly.

My issue seems firefox only and not always the case. Chrome, android and edge work fine, request the file and start playing instantly while still downloading the file.

**Reply by wardwygaerts** [reply]
I don't use firefox and only Chrome and the Immich android app. 720p is enough for me to show video's to others :)

**Reply by keesfluitman** [reply]
ye. Understandable. But the size of the files isn't the issue here. Thanks anyway! gl
